### PR TITLE
Test calculate LOD function

### PR DESCRIPTION
**Pull request by loicmarx**
A modification of the function has been made and the function has been tested.

I hereby confirm that I have:

- [X] Tested my code on my own machine
- [X] Followed the guidelines in the [Contributing Guide](https://opencobra.github.io/cobratoolbox/docs/contributing.html)
- [X] Selected `develop` as a target branch (top left drop-down menu)

*(Note: You may replace [ ] with [X] to check the box)*

### CODE REVIEW

File: src/dataIntegration/metabotools/calculateLODs.m [hunk 1]
```diff
@@ -14,14 +14,30 @@
...
+    
+end    
 
-lodmM =[];
+if length(lod_ngmL) ~= length(theo_mass)
+    
+    error('The length of the inputs are not the same');
+end
+
+lod_gL = lod_ngmL * 0.000001;
```
Comment (tpfau) [inline src/dataIntegration/metabotools/calculateLODs.m]: I would suggest to change this to:
```
lod_gL = lod_ngmL * 1e-6
```
for readability.

File: src/dataIntegration/metabotools/calculateLODs.m [hunk 2]
```diff
@@ -14,14 +14,30 @@
...
+%  Modified by Loic Marx, November 2018
 
-lod_gL = lod_ngmL* 0.000001;
+% input checking
+if length(lod_ngmL) == 1
+    
+    lod_ngmL = repmat(lod_ngmL, 1, length(theo_mass)); 
+    
+    % throw a warning
+    warning('The inputs have different size but we managed to fix this problem');
```
Comment (tpfau) [inline src/dataIntegration/metabotools/calculateLODs.m]: I'd suggest to remove the warning and update the documentation, that the inputs can either be a single value which is used for every lod or a vector of individual lods.

File: src/dataIntegration/metabotools/calculateLODs.m [hunk 3]
```diff
@@ -14,14 +14,30 @@
...
+    
+end    
 
-lodmM =[];
+if length(lod_ngmL) ~= length(theo_mass)
```
Comment (tpfau) [inline src/dataIntegration/metabotools/calculateLODs.m]: what you are looking for here is not `length` but `numel`. 
`[2,2;2,2]` and `[2,1]` have the same length but are still invalid inputs.

File: test/verifiedTests/dataIntegration/testMetaboTools/testCalculateLODs.m [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+calculateLOD = calculateLODs(theo_mass,lod_ngmL);
+
+% Comparison between calculate value and ref value
+
+assert(isequal(calculateLOD, v_ref))
+
+% Throwing an error if the dimensions of both inputs are not equal
+lod_ngmL_unequal = [1, 2, 3];
+
+assert(verifyCobraFunctionError('calculateLODs', 'inputs', {theo_mass,lod_ngmL_unequal'}, 'testMessage', 'both inputs do not have the same size'));
```
Comment (tpfau) [inline test/verifiedTests/dataIntegration/testMetaboTools/testCalculateLODs.m]: our tested message is not the same as the error message thrown. 
This should be  `'The length of the inputs are not the same'`, and I would actually suggest to improve that message:
`'The number of elements in the input vectors do not match. They have to be either the same size, or lod_ngmL has to be a single value which is used for all elements'`

File: src/dataIntegration/metabotools/calculateLODs.m [hunk 5]
```diff
@@ -14,26 +14,30 @@
 %   lod_mM:            Detection limits in mM
 %
 % .. Author: - Maike K. Aurich 27/05/15
-%  Modified by Loic Marx
+%  Modified by Loic Marx, November 2018
```
Comment (laurentheirendt) [inline src/dataIntegration/metabotools/calculateLODs.m]: This is not properly formatted.

File: src/dataIntegration/metabotools/calculateLODs.m (hunk 2 above)
Comment (laurentheirendt) [reply src/dataIntegration/metabotools/calculateLODs.m]: Sounds good. @loicmarx, may you do that? 👍

File: test/verifiedTests/dataIntegration/testMetaboTools/testCalculateLODs.m [hunk 6]
```diff
@@ -9,34 +9,45 @@
...
 currentDir = pwd;
 
-% initialize the test
+% Initialize the test
 fileDir = fileparts(which('testcalculateLODs'));
+
 cd(fileDir);
 
-% calculate reference value
+% Defining the inputs
```
Comment (laurentheirendt) [inline test/verifiedTests/dataIntegration/testMetaboTools/testCalculateLODs.m]: `% Define the inputs`

File: test/verifiedTests/dataIntegration/testMetaboTools/testCalculateLODs.m [hunk 7]
```diff
@@ -9,34 +9,45 @@
...
 v_ref = 1e-3 * [1; 0.5];
 
-%calculate inputs
-lod_ngmL = 1;
+% Calculate the actual value by using the function
 
 calculateLOD = calculateLODs(theo_mass,lod_ngmL);
-% Comparison to ref value
+
+% Comparison between calculate value and ref value
```
Comment (laurentheirendt) [inline test/verifiedTests/dataIntegration/testMetaboTools/testCalculateLODs.m]: `% Compare between the calculated value and the reference value`

File: test/verifiedTests/dataIntegration/testMetaboTools/testCalculateLODs.m [hunk 8]
```diff
@@ -9,34 +9,45 @@
...
 
 assert(isequal(calculateLOD, v_ref))
-% throwing an error if the dimensions are not equal
+
+% Throwing an error if the dimensions of both inputs are not equal
```
Comment (laurentheirendt) [inline test/verifiedTests/dataIntegration/testMetaboTools/testCalculateLODs.m]: `Throw ...`

File: test/verifiedTests/dataIntegration/testMetaboTools/testCalculateLODs.m [hunk 9]
```diff
@@ -9,34 +9,45 @@
...
 lod_ngmL_unequal = [1, 2, 3];
 
 assert(verifyCobraFunctionError('calculateLODs', 'inputs', {theo_mass,lod_ngmL_unequal'}, 'testMessage', 'both inputs do not have the same size'));
-% Testing the case where lod_ngmL is a vector of the same size as theo_mass
 
+% If both inputs are two vectors of the same size
```
Comment (laurentheirendt) [inline test/verifiedTests/dataIntegration/testMetaboTools/testCalculateLODs.m]: This comment is not very clear.

File: test/verifiedTests/dataIntegration/testMetaboTools/testCalculateLODs.m [hunk 10]
```diff
@@ -9,34 +9,45 @@
...
 lod_ngmL_vector = [1, 2];
+
 v_ref_vector = 1e-3 * [1; 1];
-% Comparison to the ref value
+
+% Comparison between the values calculated when the inputs have the same size and the reference value
```
Comment (laurentheirendt) [inline test/verifiedTests/dataIntegration/testMetaboTools/testCalculateLODs.m]: `Compare ... `

File: src/dataIntegration/metabotools/calculateLODs.m (hunk 3 above)
Comment (laurentheirendt) [reply src/dataIntegration/metabotools/calculateLODs.m]: I prefer `size()`...

Comment (tpfau) [reply src/dataIntegration/metabotools/calculateLODs.m]: Yes, but.
Size looks into both dimensions, at which point you have to either make a complicated statement or you only allow input that is either a column or a row vector.

**Comment by laurentheirendt** [comment]
Thanks @loicmarx 👍

**Pull request merged**